Sources/SwiftProtobuf/BinaryDecoder.swift

```diff
@@ -18,1477 +18,1505 @@
+                  consumed = true
+            default: return
+            }
+      }
+
+      internal mutating func decodeSingularEnumField<E: Enum>(value: inout E?)
+            throws where E.RawValue == Int {
+            guard fieldWireFormat == WireFormat.varint else { return }
             let varint = try decodeVarint()
             if let v = E(rawValue: Int(Int32(truncatingIfNeeded: varint))) {
-                value.append(v)
-                consumed = true
+                  value = v
+                  consumed = true
             }
-        case WireFormat.lengthDelimited:
-            var n: Int = 0
-            var extras: [Int32]?
-            let p = try getFieldBodyBytes(count: &n)
-            let ints = Varint.countVarintsInBuffer(start: p, count: n)
-            value.reserveCapacity(value.count + ints)
-            var subdecoder = BinaryDecoder(forReadingFrom: p, count: n, parent: self)
-            while !subdecoder.complete {
-                let u64 = try subdecoder.decodeVarint()
-                let i32 = Int32(truncatingIfNeeded: u64)
-                if let v = E(rawValue: Int(i32)) {
-                    value.append(v)
-                } else if !options.discardUnknownFields {
-                    if extras == nil {
-                        extras = []
-                    }
-                    extras!.append(i32)
-                }
+      }
+
+      internal mutating func decodeSingularEnumField<E: Enum>(value: inout E)
+            throws where E.RawValue == Int {
+            guard fieldWireFormat == WireFormat.varint else { return }
+            let varint = try decodeVarint()
+            if let v = E(rawValue: Int(Int32(truncatingIfNeeded: varint))) {
+                  value = v
+                  consumed = true
             }
-            if let extras = extras {
-                let fieldTag = FieldTag(fieldNumber: fieldNumber, wireFormat: .lengthDelimited)
-                var bodySize = 0
-                for v in extras {
-                    bodySize += Varint.encodedSize(of: Int64(v))
-                }
-                let fieldSize = Varint.encodedSize(of: fieldTag.rawValue) + Varint.encodedSize(of: Int64(bodySize)) + bodySize
-                var field = Data(count: fieldSize)
-                field.withUnsafeMutableBytes { (pointer: UnsafeMutablePointer<UInt8>) in
-                    var encoder = BinaryEncoder(forWritingInto: pointer)
-                    encoder.startField(tag: fieldTag)
-                    encoder.putVarInt(value: Int64(bodySize))
-                    for v in extras {
-                        encoder.putVarInt(value: Int64(v))
-                    }
-                }
-                unknownOverride = field
+      }
+
+      internal mutating func decodeRepeatedEnumField<E: Enum>(value: inout [E])
+            throws where E.RawValue == Int {
+            switch fieldWireFormat {
+            case WireFormat.varint:
```

Comment:
Lots of people are unhappy with dedented case's.  Personally I appreciate the way they prevent `switch`es from eating up whitespace, but for example, I couldn't convince the people on my current project that they were OK.

Reply:
TBH indenting them would remove another special case, so I wouldn't be heartbroken if folks wanted to do that. I mainly kept it because it's what Xcode does when formatting-while-typing (it initially indent when you type the `{` and move to the next line, but once you type the `:`, the clause jumps back), and I didn't want our developers to constantly be fighting against Xcode. (That's never a winnable battle though, and there are plenty of other places in our style where fighting occurs.)

Reply:
Once we get this thing out there as the official formatter, I think (without evidence—personal guess and not a statement on behalf of Apple) we'll be able to get Xcode to cooperate with it.  I sure as heck hope so.  I'll see what strings can be pulled on this end.